### PR TITLE
feat(connector): [Multisafepay] Currency Unit Conversion

### PR DESCRIPTION
**Pull request by shivam-jainn**
## Type of Change

- [ ] Bugfix
- [x ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Addressing Issue: https://github.com/juspay/hyperswitch/issues/2232
Modified two files in hyperswitch/crates/router/src/connector/
multisafepay.rs
Implement get_currency_unit function
Modify ConnectorIntegration implementations for Multisafepay

multisafepay/transformers.rs
Implement MultisafepayRouterData<T> structure and functionality
Modify implementation for MultisafepayPaymentsRequest

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

We need to create a payment using Multisafepay and test the currency unit from connector response and connector dashboard.

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ x] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/multisafepay.rs [hunk 1]
```diff
@@ -257,7 +261,13 @@ impl ConnectorIntegration<api::Authorize, types::PaymentsAuthorizeData, types::P
...
         req: &types::PaymentsAuthorizeRouterData,
     ) -> CustomResult<Option<types::RequestBody>, errors::ConnectorError> {
-        let req_obj = multisafepay::MultisafepayPaymentsRequest::try_from(req)?;
+        let connector_router_data = multisafepay::MultisafepayRouterData::try_from((
+            &self.get_currency_unit(),
+            req.request.currency,
+            req.request.amount,
+            req,
+        ))?;
+        let req_obj = multisafepay::MultisafepayPaymentsRequest::try_from(&connector_router_data)?;
```
Comment (SamraatBansal) [inline crates/router/src/connector/multisafepay.rs:270]: Hey @shivam-jainn 
You have to similarly do for the `Refunds` function that you can find in `line 950`.
There you would be using `req.request.refund_amount`

File: crates/router/src/connector/multisafepay/transformers.rs [hunk 2]
```diff
@@ -481,9 +517,9 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for MultisafepayPaymentsReques
         Ok(Self {
             payment_type,
             gateway,
-            order_id: item.connector_request_reference_id.to_string(),
-            currency: item.request.currency.to_string(),
-            amount: item.request.amount,
+            order_id: item.router_data.connector_request_reference_id.to_string(),
+            currency: item.router_data.request.currency.to_string(),
+            amount: item.amount,
```
Comment (SamraatBansal) [inline crates/router/src/connector/multisafepay/transformers.rs:522]: Similarly you have to do for `RefundRequest` in line number 672, there also we have to use `item.amount`

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

**Comment by shivam-jainn** [comment]
@SamraatBansal hey 
I have done the required changes .
 Please do guide me if there is something else that I might need to do .
Thank you

**Review by SamraatBansal — APPROVED** [review]

**Review by srujanchikke — APPROVED** [review]
Looks good to me!
Thanks for the PR @shivam-jainn

**Comment by deepanshu-iiitu** [comment]
Hey @shivam-jainn  ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Pull request merged**